plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -119,7 +119,12 @@ public void execute(IngestDocument ingestDocument) {
 
             if (properties.contains(Property.CONTENT_LENGTH)) {
                 String contentLength = metadata.get(Metadata.CONTENT_LENGTH);
-                String length = Strings.hasLength(contentLength) ? contentLength : String.valueOf(parsedContent.length());
+                long length;
+                if (Strings.hasLength(contentLength)) {
+                    length = Long.parseLong(contentLength);
```

Comment:
Hmm...shouldn't we? Isn't no leniency pretty much our modus operandi? And there's the ability to attach an on error handler?